routers/api/v1/org/avatar.go

```diff
@@ -43,12 +47,25 @@ func UpdateAvatar(ctx *context.APIContext) {
 		return
 	}
 
-	err = user_service.UploadAvatar(ctx, ctx.Org.Organization.AsUser(), content)
+	user := ctx.Org.Organization.AsUser()
+	hasAvatar := user.HasAvatar()
+
+	avatarData, err := user_service.UploadAvatar(ctx, user, content)
 	if err != nil {
 		ctx.Error(http.StatusInternalServerError, "UploadAvatar", err)
+		return
 	}
 
-	ctx.Status(http.StatusNoContent)
+	if hasAvatar {
+		ctx.Status(http.StatusOK)
+	} else {
+		ctx.Status(http.StatusCreated)
+	}
+	_, err = ctx.Write(avatarData)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "Write", err)
+		return
+	}
```

Comment:
Why does the upload now respond with the avatar?

Reply:
https://github.com/go-gitea/gitea/pull/29185#issuecomment-1950910348

Mdn specification is written `the newly created items are returned in the body of the message` for 201

Reply:
Where?
> [201 Created](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201)
> The request succeeded, and a new resource was created as a result. This is typically the response sent after POST requests, or some PUT requests.

I would not expect a server to responsd with my 1gb file I just uploaded. I understand it for endpoints like creating a user to return the user json but not for file uploads.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

Reply:
Maybe don't return any values like before?